celery/backends/elasticsearch.py

```diff
@@ -53,6 +58,17 @@ def __init__(self, url=None, *args, **kwargs):
         self.host = host or self.host
         self.port = port or self.port
 
+        self.es_retry_on_timeout = (
+                _get('elasticsearch_retry_on_timeout') or self.es_retry_on_timeout
+                )
+        self.es_timeout = (
+                _get('elasticsearch_timeout') or self.es_timeout
+                )
+
+        self.es_max_retries = (
+                _get('elasticsearch_max_retries') or self.es_max_retries
```

Comment:
If someone sets maximum retries to 0, the setting value will revert to the default. So, perhaps you need to check explicitly for `None`?